### DISCUSSION TITLE
Steps and Tasks to deploy dev tested zipline services on live vps behind cdn ?

### DISCUSSION THREAD

**Discussion by Shabinder** [cashapp/zipline#1094] [Q&A] [answered]
- Project Readme is pretty clear on how to get started and test your zipline services, using tasks like
   - :serveProductionZipline
   - :serveDevelopmentZipline

Now after doing above, instruction or not pretty clear on how to host these on a server behind a cdn.
I dont assume we will have a gradle running serving using above commands. so whats the best setup to have.

Where to copy the binaries from? , link to those maybe using a server like nginx or apache from vpn behind a cdn ?

would appreciate any guidance / insights you guys put light to.
thanks.

**Comment by Shabinder** [comment]
Things I found:
 - on running _compileProductionExecutableKotlinJsZipline_ ,  a folder in `build/compileSync/js/main/productionExecutable/kotlinZipline` with **manifest.zipline.json** and ***.zipline** binaries are generated.
 - If I host these statically from a vps say from a folder.
 - Now if I want to host multiple plugins with same deps, can I just put all manifests in same folder and host ? ( _kotlin deps zipline binaries will overwrite since all are same_ , **_that will be an issue or not ?_**)

**Comment by swankjesse** [comment, marked as answer]
We upload that production build directory to a CDN, using a git commit SHA as the directory name.

We use LaunchDarkly to configure which directory client apps get code from.

We have an automation in CI that coordinates signing code, uploading to CDN, creating variants in LaunchDarkly, and promoting which variants are active in staging and production.

Unfortunately our tools are very custom and not open source.

**Reply by erawhctim** [reply]
How do you handle versioning the code on the CDN across app versions? Does your LD configuration specify all of that, and you just determine if/when a new app release needs to be associated with a new JS deploy to the CDN?
